liblocation/src/main/java/com/mapbox/android/core/location/LocationCallbackTransport.java

```diff
@@ -13,30 +17,62 @@
  */
 public class LocationCallbackTransport implements LocationListener {
   private static final String TAG = "CallbackTransport";
+  private static final int MESSAGE_WHAT = 0;
   protected final LocationEngineCallback<LocationEngineResult> callback;
   private Location currentBestLocation;
-  private long lastUpdateTime = 0;
   private long fastestInterval = 0;
+  private Location cacheLocation;
+  private Handler handler;
+  private HandlerThread handlerThread;
 
   LocationCallbackTransport(LocationEngineCallback<LocationEngineResult> callback) {
     this.callback = callback;
+    handlerThread = new HandlerThread(this.getClass().getName());
```

Comment:
Why do we need a separate thread here? It forces the location update delivery on a background thread which we should avoid at all cost. It should be just fine to process the `handleMessage` logic on the main thread.